### PR TITLE
[BUFGIX BETA] dont run feature flag model tests in beta

### PR DESCRIPTION
**Pull request by runspired**
Master version of #6986

### CODE REVIEW

File: packages/unpublished-model-encapsulation-test-app/ember-cli-build.js [hunk 1]
```diff
@@ -5,8 +5,20 @@
 const EmberApp = require('ember-cli/lib/broccoli/ember-app');
 
 module.exports = function(defaults) {
+  const isProd = process.env.EMBER_ENV === 'production';
+  const compatWith = process.env.EMBER_DATA_FULL_COMPAT ? '99.0' : null;
```
Comment (igorT) [inline packages/unpublished-model-encapsulation-test-app/ember-cli-build.js]: Whats this about? Can you please add a comment

File: packages/unpublished-model-encapsulation-test-app/package.json [hunk 2]
```diff
@@ -18,6 +18,7 @@
     "test": "ember test --test-port=0"
   },
   "dependencies": {
+    "@ember-data/private-build-infra": "3.17.0-alpha.1",
```
Comment (igorT) [inline packages/unpublished-model-encapsulation-test-app/package.json]: Do you need to add `ember-data/canary-features` in devDependencies at least?

**Comment by igorT** [comment]
A better title might be > dont run feature flagged tests in beta

File: packages/unpublished-model-encapsulation-test-app/ember-cli-build.js (hunk 1 above)
Comment (runspired) [reply packages/unpublished-model-encapsulation-test-app/ember-cli-build.js]: this ensures deprecation stripping when desired as well, matching the config and flag used by the tests in -ember-data package

File: packages/unpublished-model-encapsulation-test-app/package.json (hunk 2 above)
Comment (runspired) [reply packages/unpublished-model-encapsulation-test-app/package.json]: we don't have to, as it is a transitive dep of all of these other things, but making it explicit can't hurt

**Pull request merged**